### DISCUSSION TITLE
Task Name in navbar

### DISCUSSION THREAD

**Discussion by flightmansam** [minbrowser/min#1976] [Show and tell]
I sometimes find that switching between tasks using the keyboard shortcuts means I have to figure out which task I'm in by looking at the tabs that are there. I quickly implemented some text that will tell you the name of the task you are in. 

Main reasons for doing this:
- maintaining task consistency: 
    - rather than firing up min and searching away with no regard for the task I am in I can quickly see whether the task I am in is appropriate and if not, switch to a one that is (or even create a new named task with the bang)
- more situational awareness when switching tasks with keyboard commands (and then using the task overlay for more intertask organisation/deleting irrelevant tabs/collapsing tasks/[exporting tasks](https://github.com/minbrowser/min/issues/1861) (WIP))

Arguably this creates more clutter in the UI. It is something that doesn't have to always be there:
- perhaps it could only appear when there are certain modifiers held down (ctrl+option, for example), or
- the text would appear for a time when you have clicked any of the navbar items, or
- would appear briefly when cycling between tasks using the keyboard shortcuts

As it is [currently implemented](https://github.com/flightmansam/min/tree/315d835d414f1caae7c5a8b6bc009449abcd6ad3/) on my fork:

https://user-images.githubusercontent.com/34013162/164192135-df502852-dac2-44b5-9d68-9bbbdecf3f96.mov

It also respects focus mode (I'm not sure how many people use focus mode?)

https://user-images.githubusercontent.com/34013162/164193947-ded5804b-84dc-4dab-bbfa-5bd15130f62a.mov

Things left to do:
- I'm not sure I nailed the padding of the text, it looks like the right side of the text is closer to the task overlay button than the left side of the text is to the "+" button. I am not a very good CSS warrior.
- Some kind of decluttering logic as mentioned above.

**Comment by PalmerAL** [comment]
I think the task name looks kind of weird as is. Maybe if the task name and the menu button had a rounded rect around them and a light background, so that they were all one button, it would look better? But even then, it seems like it's adding more clutter to the navbar (and a lot of people don't use tasks at all, so just having a permanent "task n" sitting in the corner is going to be annoying).

I agree there should be some way to see what task you're switching to when using the keyboard shortcuts. Perhaps even better would be a ppup in the center of the screen that shows the new task name, and fades out after you lift up the keys. Although that's a little difficult to implement, because you have to stack it on top of the browserview as a popup window.
